crates/matrix-sdk-crypto/src/types/mod.rs

```diff
@@ -43,6 +43,9 @@ pub enum Signature {
     Ed25519(Ed25519Signature),
     /// An unknown digital signature as a base64 encoded string.
     Other(String),
+    /// An invalid signature that could not be decoded, left unmodified as a
+    /// string.
+    Invalid(String),
```

Comment:
I'm concerned the intent here is insufficiently clear. The intent being that `Other` represents a signature in an *unknown* (not supported by the Rust SDK) algorithm, while `Invalid` represents a signature claimed to be in a *known* algorithm, but where the payload doesn't have the right format for that algorithm.

We also need to make it explicit that `Signature` does *not* represent a *validated* signature, just a potentially decoded signature.

I'll try to cook up a concrete suggestion.

Reply:
Though after typing that out, I feel like just using `Result` instead of the `Invalid` case would be better. The doc improvement suggestion still stands (but without the case describing `Invalid`).